geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseIndexAdapter.scala

```diff
@@ -370,13 +347,14 @@ class HBaseIndexAdapter(ds: HBaseDataStore) extends IndexAdapter[HBaseDataStore]
 
       def addGroup(group: java.util.List[RowRange]): Unit = {
         val s = new Scan(group.get(0).getStartRow, group.get(group.size() - 1).getStopRow).addFamily(colFamily)
-        if (group.size() > 1) {
+        // note: coprocessors always expect a filter list
+        if (coprocessor || group.size() > 1) {
           // TODO GEOMESA-1806
           // currently, the MultiRowRangeFilter constructor will call sortAndMerge a second time
           // this is unnecessary as we have already sorted and merged
           val mrrf = new MultiRowRangeFilter(group)
           // note: mrrf first priority
-          s.setFilter(if (filters.isEmpty) { mrrf } else { new FilterList(filters.+:(mrrf): _*) })
+          s.setFilter(if (coprocessor || filters.nonEmpty) { new FilterList(filters.+:(mrrf): _*) } else { mrrf })
```

Comment:
Ok, ok, should we talk about this code + comments now?  Or should I merge all the things and then we sort this out? 

I'm trying to figure out when to grok this so that we can dial in the partial results.

Reply:
this pr has some partial changes to the coprocessor, which is cleaned up in the follow-on pr, so maybe that would be a better place to chat about that type of thing. but here is fine too, i think the basic concepts are the same